### PR TITLE
Vim: Bump version

### PR DESCRIPTION
**Pull request by bitigchi**
This PR updates Vim to 8.2.0624.

### CODE REVIEW

**Comment by bitigchi** [comment]
Please do not merge this yet as gVim crashes when entering ex commands. See #4899.

**Comment by waddlesplash** [comment]
Was the patch really upstreamed?

**Comment by bitigchi** [comment]
@waddlesplash yes. I will bump the version once fix for #4899 gets merged as well.

Bonus: See the [readme](https://github.com/vim/vim) 😉.

**Comment by bitigchi** [comment]
This one's ready to merge.

**Comment by Begasus** [comment]
Line 59, add a tab before the "

**Comment by Begasus** [comment]
Building on gcc2h fails with:
```
/packages/gcc-2.95.3_2017_07_20-2/.self/develop/tools/i586-pc-haiku/bin/ld: cannot find -lsupc++
```
libsubc++ isn't provided by gcc2

**Comment by Begasus** [comment]
Seems to be OK if you add secondary architecture in the recipe though: https://ibb.co/gSSyfph

**Comment by Begasus** [comment]
OK, check again against gcc2, atleast this patch needs to be applied (2 lines)
https://github.com/haikuports/haikuports/blob/38c09865556599fd72662eb21dac3d5e20337a6e/app-editors/vim/patches/vim-8.1.1240.patchset#L6099

File: app-editors/vim/vim-8.2.0639.recipe [hunk 1]
```diff
@@ -10,72 +10,78 @@ configuration files.
...
-REVISION="4"
+REVISION="1"
 SOURCE_URI="https://github.com/vim/vim/archive/v$portVersion.tar.gz"
-CHECKSUM_SHA256="3bfb6b0de36e96862933e49df93412a46a0914356ec16dd60a5ccbd5cdc46d25"
+CHECKSUM_SHA256="5ec230131a9dff131fa199f47fd391341625d0bad056aa3eed970b70628886ae"
 SOURCE_FILENAME="vim-$portVersion.tar.gz"
-PATCHES="vim-$portVersion.patchset"
 
-ARCHITECTURES="x86_gcc2 ?x86 x86_64"
+ARCHITECTURES="x86_gcc2 x86_64"
```
Comment (Begasus) [inline app-editors/vim/vim-8.2.0639.recipe:20]: Without patching it won't work for x86_gcc2 so:
```diff
- ARCHITECTURES="x86_gcc2 x86_64"
+ ARCHITECTURES="!x86_gcc2 x86_64"
```

Comment (bitigchi) [reply app-editors/vim/vim-8.2.0639.recipe]: Fixed!

File: app-editors/vim/vim-8.2.0639.recipe [hunk 2]
```diff
@@ -144,15 +149,8 @@ BUILD()
...
-	cat << EOF > $dataDir/vim/vimrc
-" Workaround to get the themes and plugins work
-execute "set runtimepath+=" . substitute(system("finddir B_USER_SETTINGS_DIRECTORY"), '\n\+$', '', '') . "/vim"
-EOF
-
-	ln -s vim $binDir/vi
-
-	addAppDeskbarSymlink $binDir/gvim Vim
+	ln -s vim $commandBinDir/vi
+	addAppDeskbarSymlink $commandBinDir/gvim Vim
```
Comment (Begasus) [inline app-editors/vim/vim-8.2.0639.recipe:153]: As the GUI window shows "VIM" maybe:
```diff
- 	addAppDeskbarSymlink $commandBinDir/gvim Vim
+ 	addAppDeskbarSymlink $commandBinDir/gvim VIM
```

Comment (bitigchi) [reply app-editors/vim/vim-8.2.0639.recipe:153]: The official name is Vim.

Comment (Begasus) [reply app-editors/vim/vim-8.2.0639.recipe:153]: Doesn't show in the Window :) https://ibb.co/7pmx65Y

**Pull request merged**